src/python/grpcio/grpc/_cython/_cygrpc/call.pyx.pxi

```diff
@@ -106,6 +107,8 @@ cdef class Call:
   def __dealloc__(self):
     if self.c_call != NULL:
       grpc_call_destroy(self.c_call)
+    grpc_shutdown()
```

Comment:
From Cython [docs about compatibility with PyPy](http://docs.cython.org/en/latest/src/userguide/pypy.html)
"a deallocator method of an extension type (__dealloc__()) may get called at a much later point than in CPython, triggered rather by memory getting tighter than by objects dying."
and
"If the point in the code is known when an object is supposed to die (e.g. when it is tied to another object or to the execution time of a function), it is worth considering if it can be invalidated and cleaned up manually at that point, rather than relying on a deallocator."